backend/webapi/Features/EndorsementRequests/Create.cs

```diff
@@ -93,24 +114,26 @@ public async Task<Model> HandleAsync(Command command)
             this.context.EndorsementRequests.Add(request);
             await this.context.SaveChangesAsync();
 
-            await this.SendEndorsementRequestEmailAsync(request.RecipientEmail, request.Token, partyName);
+            await this.SendEndorsementRequestEmailAsync(request.RecipientEmail, request.Token, partyName, isUnlicensed);
 
             return new(false);
         }
 
-        private async Task SendEndorsementRequestEmailAsync(string recipientEmail, Guid token, string partyName)
+        private async Task SendEndorsementRequestEmailAsync(string recipientEmail, Guid token, string partyName, bool isUnlicensed)
         {
             string url = this.applicationUrl.SetQueryParam("endorsement-token", token);
             var link = $"<a href=\"{url}\" target=\"_blank\" rel=\"noopener noreferrer\">this link</a>";
             var pidpSupportEmail = $"<a href=\"mailto:{EmailService.PidpEmail}\">{EmailService.PidpEmail}</a>";
             var pidpSupportPhone = $"<a href=\"tel:{EmailService.PidpSupportPhone}\">{EmailService.PidpSupportPhone}</a>";
+            var templatetext = isUnlicensed ? "You are receiving this email because a user is requesting to endorse you."
```

Comment:
@Paahn  can we make the templatetext as environmental variable and inject it through infrastructure as a code in server.
So that if we want to make any changes to the email template then updating the environmental variables in the deployment scripts will be enough rather than changing the application code here. It saves the time of re-build and deploy as well.

Reply:
@james-hollinger

Reply:
There are always trade offs when deciding to environmentalize text in this way. 

The main upside is that we could get changes into prod faster, without having to go through a full dev > test > prod code release. Time spent re-building the backend is negligible, and you still have to re-deploy when you change an environment variable.

There are several downside though: it is now harder to be certain about the content and formatting of your emails, since the text now lives in Openshift rather than in your code. Since prod is inaccessible to almost everyone, we also can't be as sure that all environments are behaving the same. It also introduces a human failure point when copy-pasting into Openshift. And it increases our infrastructure code and proliferates environment variables.

For many applications, that first upside outweighs any downsides. Especially for any apps that have long release cycles or have text that changes often.
PIdP has neither of those things. We release to prod often, and at a cadence set by us rather than product releases or external forces. Our text also hardly ever changes; this is only the second time this text has been requested to change and most of the other text in the app has never changed.

I understand the idea to make it easier to update text like this without having to make code changes but I don't think it is desirable here.

Reply:
I agree with James. I am concerned that the proposed change would be surprising to new developers as storing email text in a deployed environment is not expected behavior, possibly increasing complexity for making text changes.

Reply:
Thanks James for the detailed explanation.